allofplos/makedb.py

```diff
@@ -122,9 +122,7 @@ class CoAuthorPLOSArticle(BaseModel):
 allfiles = Corpus(corpus_dir).files
 files = random.sample(allfiles, args.random) if args.random else allfiles
 
-for file_ in tqdm(files):
-    doi = filename_to_doi(file_)
-    article = Article(doi)
+for article in tqdm(Corpus(corpus_dir)):
```

Comment:
if you don't iterate over "files" you are not iterating over the random sample if that is the option in the command line

Reply:
Thanks for catching that. The Corpus iterator is `random_article_generator`, so I'll solve this by limiting the number of articles we pull from Corpus.

One reason I neglected tests with this pull request is that I'm uncertain what you want to enshrine as permanent. Did you imagine the database code combining with the xml based classes, or staying as a separate utility?

Also, sorry for the lack of tests.

Reply:
I never thought of this utility as something that can be integrated with xml classes, so as a separate script is OK for me.

Reply:
waiting for the commit with "I'll solve this by limiting the number of articles we pull from Corpus."